# Exploring Ebay Car Sales Data

The aim of this project is to clean used car listings from eBay data and analyze results for useful insights.

In [1]:
import numpy as np
import pandas as pd

In [2]:
autos = pd.read_csv("autos.csv", encoding='Latin-1')

In [3]:
autos

dateCrawled                                               name  \
0      2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1      2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2      2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3      2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4      2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   
5      2016-03-21 13:47:45  Chrysler_Grand_Voyager_2.8_CRD_Aut.Limited_Sto...   
6      2016-03-20 17:55:21  VW_Golf_III_GT_Special_Electronic_Green_Metall...   
7      2016-03-16 18:55:19                               Golf_IV_1.9_TDI_90PS   
8      2016-03-22 16:51:34                                         Seat_Arosa   
9      2016-03-16 13:47:02          Renault_Megane_Scenic_1.6e_RT_Klimaanlage   
10     2016-03-15 01:41:36                       VW_Golf_Tuning_in_siber/grau   
11     2016-03-16 18:45:34                         Mercedes_A140_Motorschaden   
12     2016-03-31 19:48:22  Smart_smart_fortwo_coupe_softouch_pure_MHD_Pan...   
13     2016-03-23 10:48:32                                 Audi_A3_1.6_tuning   
14     2016-03-23 11:50:46  Renault_Clio_3__Dynamique_1.2__16_V;_viele_Ver...   
15     2016-04-01 12:06:20        Corvette_C3_Coupe_T_Top_Crossfire_Injection   
16     2016-03-16 14:59:02                                Opel_Vectra_B_Kombi   
17     2016-03-29 11:46:22                          Volkswagen_Scirocco_2_G60   
18     2016-03-26 19:57:44               Verkaufen_mein_bmw_e36_320_i_touring   
19     2016-03-17 13:36:21        mazda_tribute_2.0_mit_gas_und_tuev_neu_2018   
20     2016-03-05 19:57:31  Audi_A4_Avant_1.9_TDI_*6_Gang*AHK*Klimatronik*...   
21     2016-03-06 19:07:10                      Porsche_911_Carrera_4S_Cabrio   
22     2016-03-28 20:50:54                               MINI_Cooper_S_Cabrio   
23     2016-03-10 19:55:34   Peugeot_Boxer_2_2_HDi_120_Ps_9_Sitzer_inkl_Klima   
24     2016-04-03 11:57:02                         BMW_535i_xDrive_Sport_Aut.   
25     2016-03-21 21:56:18  Ford_escort_kombi_an_bastler_mit_ghia_ausstattung   
26     2016-04-03 22:46:28                                Volkswagen_Polo_Fox   
27     2016-03-27 18:45:01             Hat_einer_Ahnung_mit_Ford_Galaxy_HILFE   
28     2016-03-19 21:56:19                                      MINI_Cooper_D   
29     2016-04-02 12:45:44  Mercedes_Benz_E_320_T_CDI_Avantgarde_DPF7_Sitz...   
...                    ...                                                ...   
49970  2016-03-21 22:47:37  c4_Grand_Picasso_mit_Automatik_Leder_Navi_Temp...   
49971  2016-03-29 14:54:12                                         W.Lupo_1.0   
49972  2016-03-26 22:25:23          Mercedes_Benz_Vito_115_CDI_Extralang_Aut.   
49973  2016-03-27 05:32:39                   Mercedes_Benz_SLK_200_Kompressor   
49974  2016-03-20 10:52:31  Golf_1_Cabrio_Tuev_Neu_viele_Extras_alles_eing...   
49975  2016-03-27 20:51:39          Honda_Jazz_1.3_DSi_i_VTEC_IMA_CVT_Comfort   
49976  2016-03-19 18:56:05  Audi_80_Avant_2.6_E__Vollausstattung!!_Einziga...   
49977  2016-03-31 18:37:18                        Mercedes_Benz_C200_Cdi_W203   
49978  2016-04-04 10:37:14                        Mercedes_Benz_E_200_Classic   
49979  2016-03-20 18:38:40                      Volkswagen_Polo_1.6_TDI_Style   
49980  2016-03-12 10:55:54                            Ford_Escort_Turnier_16V   
49981  2016-03-15 09:38:21             Opel_Astra_Kombi_mit_Anhaengerkupplung   
49982  2016-03-29 18:51:08            Skoda_Fabia_4_Tuerer_Bj:2004__85.000Tkm   
49983  2016-03-06 12:43:04                                      Ford_focus_99   
49984  2016-03-31 22:48:48  Student_sucht_ein__Anfaengerauto___ab_2000_BJ_...   
49985  2016-04-02 16:38:23                          Verkaufe_meinen_vw_vento!   
49986  2016-04-04 20:46:02  Chrysler_300C_3.0_CRD_DPF_Automatik_Voll_Ausst...   
49987  2016-03-22 20:47:27  Audi_A3_Limousine_

In [4]:
autos.info()
autos.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50000 entries, 0 to 49999
Data columns (total 20 columns):
dateCrawled            50000 non-null object
name                   50000 non-null object
seller                 50000 non-null object
offerType              50000 non-null object
price                  50000 non-null object
abtest                 50000 non-null object
vehicleType            44905 non-null object
yearOfRegistration     50000 non-null int64
gearbox                47320 non-null object
powerPS                50000 non-null int64
model                  47242 non-null object
odometer               50000 non-null object
monthOfRegistration    50000 non-null int64
fuelType               45518 non-null object
brand                  50000 non-null object
notRepairedDamage      40171 non-null object
dateCreated            50000 non-null object
nrOfPictures           50000 non-null int64
postalCode             50000 non-null int64
lastSeen               50000 non-null obj

dateCrawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   seller offerType   price   abtest vehicleType  yearOfRegistration  \
0  privat   Angebot  $5,000  control         bus                2004   
1  privat   Angebot  $8,500  control   limousine                1997   
2  privat   Angebot  $8,990     test   limousine                2009   
3  privat   Angebot  $4,350  control  kleinwagen                2007   
4  privat   Angebot  $1,350     test       kombi                2003   

     gearbox  powerPS   model   odometer  monthOfRegistration fuelType  \
0    manuell      158  andere  150,000km                    3      lpg   
1  automatik      286     7er  150,000km                    6   benzin   
2    manuell      102    golf   70,000km                    7   benzin   
3  automatik       71  fortwo   70,000km                    6   benzin   
4    manuell        0   focus  150,000km                    7   benzin   

        brand notRepairedDamage          dateCreated  nrOfPictures  \
0     peugeot              nein  2016-03-26 00:00:00             0   
1         bmw              nein  2016-04-04 00:00:00             0   
2  volkswagen              nein  2016-03-26 00:00:00             0   
3       smart              nein  2016-03-12 00:00:00             0   
4        ford              nein  2016-04-01 00:00:00             0   

   postalCode             lastSeen  
0       79588  2016-04-06 06:45:54  
1       71034  2016-04-06 14:45:08  
2       35394  2016-04-06 20:15:37  
3       33729  2016-03-15 03:16:28  
4       39218  2016-04-01 14:38:50

- We have 20 columns in the dataset, of which 15 have no nulls (missing values). We have a few cells that can be converted to the numeric form including price and odometer. The notRepairedDamage column could be converted to boolean and the column names could be cleaned for easier reading.



In [5]:
autos.columns

Index(['dateCrawled', 'name', 'seller', 'offerType', 'price', 'abtest',
       'vehicleType', 'yearOfRegistration', 'gearbox', 'powerPS', 'model',
       'odometer', 'monthOfRegistration', 'fuelType', 'brand',
       'notRepairedDamage', 'dateCreated', 'nrOfPictures', 'postalCode',
       'lastSeen'],
      dtype='object')

In [6]:
autos.columns = ['date_crawled', 'name', 'seller', 'offer_type', 'price', 'abtest',
       'vehicle_type', 'registration_year', 'gearbox', 'power_ps', 'model',
       'odometer', 'registration_month', 'fuel_type', 'brand',
       'unrepaired_damage', 'ad_created', 'nr_of_pictures', 'postal_code',
       'last_seen']

In [7]:
autos.head()

date_crawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   seller offer_type   price   abtest vehicle_type  registration_year  \
0  privat    Angebot  $5,000  control          bus               2004   
1  privat    Angebot  $8,500  control    limousine               1997   
2  privat    Angebot  $8,990     test    limousine               2009   
3  privat    Angebot  $4,350  control   kleinwagen               2007   
4  privat    Angebot  $1,350     test        kombi               2003   

     gearbox  power_ps   model   odometer  registration_month fuel_type  \
0    manuell       158  andere  150,000km                   3       lpg   
1  automatik       286     7er  150,000km                   6    benzin   
2    manuell       102    golf   70,000km                   7    benzin   
3  automatik        71  fortwo   70,000km                   6    benzin   
4    manuell         0   focus  150,000km                   7    benzin   

        brand unrepaired_damage           ad_created  nr_of_pictures  \
0     peugeot              nein  2016-03-26 00:00:00               0   
1         bmw              nein  2016-04-04 00:00:00               0   
2  volkswagen              nein  2016-03-26 00:00:00               0   
3       smart              nein  2016-03-12 00:00:00               0   
4        ford              nein  2016-04-01 00:00:00               0   

   postal_code            last_seen  
0        79588  2016-04-06 06:45:54  
1        71034  2016-04-06 14:45:08  
2        35394  2016-04-06 20:15:37  
3        33729  2016-03-15 03:16:28  
4        39218  2016-04-01 14:38:50

- The column names used camelcase instead of Python's preferred snakecase. Changes to the column names have been made for readability of the user and computer.

In [8]:
autos.describe(include="all")

date_crawled         name  seller offer_type  price abtest  \
count                 50000        50000   50000      50000  50000  50000   
unique                48213        38754       2          2   2357      2   
top     2016-03-23 19:38:20  Ford_Fiesta  privat    Angebot     $0   test   
freq                      3           78   49999      49999   1421  25756   
mean                    NaN          NaN     NaN        NaN    NaN    NaN   
std                     NaN          NaN     NaN        NaN    NaN    NaN   
min                     NaN          NaN     NaN        NaN    NaN    NaN   
25%                     NaN          NaN     NaN        NaN    NaN    NaN   
50%                     NaN          NaN     NaN        NaN    NaN    NaN   
75%                     NaN          NaN     NaN        NaN    NaN    NaN   
max                     NaN          NaN     NaN        NaN    NaN    NaN   

       vehicle_type  registration_year  gearbox      power_ps  model  \
count         44905       50000.000000    47320  50000.000000  47242   
unique            8                NaN        2           NaN    245   
top       limousine                NaN  manuell           NaN   golf   
freq          12859                NaN    36993           NaN   4024   
mean            NaN        2005.073280      NaN    116.355920    NaN   
std             NaN         105.712813      NaN    209.216627    NaN   
min             NaN        1000.000000      NaN      0.000000    NaN   
25%             NaN        1999.000000      NaN     70.000000    NaN   
50%             NaN        2003.000000      NaN    105.000000    NaN   
75%             NaN        2008.000000      NaN    150.000000    NaN   
max             NaN        9999.000000      NaN  17700.000000    NaN   

         odometer  registration_month fuel_type       brand unrepaired_damage  \
count       50000        50000.000000     45518       50000             40171   
unique         13                 NaN         7          40                 2   
top     150,000km                 NaN    benzin  volkswagen              nein   
freq        32424                 NaN     30107       10687             35232   
mean          NaN            5.723360       NaN         NaN               NaN   
std           NaN            3.711984       NaN         NaN               NaN   
min           NaN            0.000000       NaN         NaN               NaN   
25%           NaN            3.000000       NaN         NaN               NaN   
50%           NaN            6.000000       NaN         NaN               NaN   
75%           NaN            9.000000       NaN         NaN               NaN   
max           NaN           12.000000       NaN         NaN               NaN   

                 ad_created  nr_of_pictures   postal_code            last_seen  
count                 50000         50000.0  50000.000000                50000  
unique                   76             NaN           NaN                39481  
top     2016-04-03 00:00:00             NaN           NaN  2016-04-07 06:17:27  
freq                   1946             NaN           NaN                    8  
mean                    NaN             0.0  50813.627300                  NaN  
std                     NaN             0.0  25779.747957                  NaN  
min                     NaN             0.0   1067.000000                  NaN  
25%                     NaN             0.0  30451.000000                  NaN  
50%                     NaN             0.0  49577.000000                  NaN  
75%                     NaN             0.0  71540.000000                  NaN  
max                     NaN             0.0  99998.000000                  NaN

- The nr_of_pictures column has only zeroes in it and so can be dropped as it is not useful. Also, the seller and offer_type are almost completely the same for each listing (private and Angebot, respectively) so these columns can be dropped as well.
- The maximum and minimum values for registration_year, power_ps seem to be weird and should be investigated for correct information and/or outliers.
- As noted earlier, the odometer and price columns have numeric data that should be converted to numeric.
- First we drop the three columns we deemed not useful.

In [9]:
autos = autos.drop(['nr_of_pictures','seller','offer_type'],axis = 1)

In [10]:
autos['odometer'] = (autos['odometer']
                         .str.replace(',','')
                         .str.replace('km','')
                         .astype(int)
                    )

autos['price'] = (autos['price']
                      .str.replace('$','')
                      .str.replace(',','')
                      .astype(int)
                  )

autos.rename(columns = {"odometer": "odometer_km"}, inplace=True)
autos.rename(columns = {"price": "price_dollars"}, inplace=True)

In [11]:
autos["odometer_km"].unique().shape

(13,)

In [12]:
autos["odometer_km"].describe()

count     50000.000000
mean     125732.700000
std       40042.211706
min        5000.000000
25%      125000.000000
50%      150000.000000
75%      150000.000000
max      150000.000000
Name: odometer_km, dtype: float64

In [13]:
autos["odometer_km"].value_counts().sort_index(ascending=False).head()

150000    32424
125000     5170
100000     2169
90000      1757
80000      1436
Name: odometer_km, dtype: int64

In [14]:
autos["price_dollars"].unique().shape

(2357,)

In [15]:
autos["price_dollars"].describe()

count    5.000000e+04
mean     9.840044e+03
std      4.811044e+05
min      0.000000e+00
25%      1.100000e+03
50%      2.950000e+03
75%      7.200000e+03
max      1.000000e+08
Name: price_dollars, dtype: float64

In [16]:
autos["price_dollars"].value_counts().sort_index()

0           1421
1            156
2              3
3              1
5              2
8              1
9              1
10             7
11             2
12             3
13             2
14             1
15             2
17             3
18             1
20             4
25             5
29             1
30             7
35             1
40             6
45             4
47             1
49             4
50            49
55             2
59             1
60             9
65             5
66             1
            ... 
151990         1
155000         1
163500         1
163991         1
169000         1
169999         1
175000         1
180000         1
190000         1
194000         1
197000         1
198000         1
220000         1
250000         1
259000         1
265000         1
295000         1
299000         1
345000         1
350000         1
999990         1
999999         2
1234566        1
1300000        1
3890000        1
10000000       1
11111111       2
12345678      

In [17]:
autos[autos['price_dollars'] > 200000].sort_values(by='price_dollars')

date_crawled                                               name  \
37840  2016-03-21 10:50:12                                        Porsche_997   
38299  2016-03-28 22:25:25                                Glas_BMW_mit_Wasser   
47337  2016-04-05 10:25:38                                    BMW_Z8_roadster   
12682  2016-03-28 22:48:01  Porsche_GT3_RS__PCCB__Lift___grosser_Exklusiv_...   
35923  2016-04-03 07:56:23  Porsche_911_Targa_Exclusive_Edition__1_von_15_...   
34723  2016-03-23 16:37:29  Porsche_Porsche_911/930_Turbo_3.0__deutsche_Au...   
14715  2016-03-30 08:37:24                 Rolls_Royce_Phantom_Drophead_Coupe   
36818  2016-03-27 18:37:37                                        Porsche_991   
37585  2016-03-29 11:38:54                                Volkswagen_Jetta_GT   
43049  2016-03-21 19:53:52                                      2_VW_Busse_T3   
514    2016-03-17 09:53:08                   Ford_Focus_Turnier_1.6_16V_Style   
22947  2016-03-22 12:54:19                         Bmw_530d_zum_ausschlachten   
7814   2016-04-04 11:53:31                                        Ferrari_F40   
47634  2016-04-04 21:25:21                                        Ferrari_FXX   
11137  2016-03-29 23:52:57  suche_maserati_3200_gt_Zustand_unwichtig_laufe...   
24384  2016-03-21 13:57:51                            Schlachte_Golf_3_gt_tdi   
2897   2016-03-12 21:50:57   Escort_MK_1_Hundeknochen_zum_umbauen_auf_RS_2000   
27371  2016-03-09 15:45:47                                         Fiat_Punto   
39377  2016-03-08 23:53:51                        Tausche_volvo_v40_gegen_van   
47598  2016-03-31 18:56:54  Opel_Vectra_B_1_6i_16V_Facelift_Tuning_Showcar...   
42221  2016-03-08 20:39:05                                  Leasinguebernahme   
39705  2016-03-22 14:58:27                        Tausch_gegen_gleichwertiges   

       price_dollars   abtest vehicle_type  registration_year    gearbox  \
37840         220000     test        coupe               2008    manuell   
38299         250000     test          NaN               2015        NaN   
47337         259000     test       cabrio               2001    manuell   
12682         265000  control        coupe               2016  automatik   
35923         295000     test       cabrio               2015  automatik   
34723         299000     test        coupe               1977    manuell   
14715         345000  control       cabrio               2012  automatik   
36818         350000  control        coupe               2016    manuell   
37585         999990     test    limousine               1985    manuell   
43049         999999     test          bus               1981    manuell   
514           999999     test        kombi               2009    manuell   
22947        1234566  control        kombi               1999  automatik   
7814         1300000  control        coupe               1992        NaN   
47634        3890000     test        coupe               2006        NaN   
11137       10000000  control        coupe               1960    manuell   
24384       11111111     test          NaN               1995        NaN   
2897        11111111     test    limousine               1973    manuell   
27371       12345678  control          NaN               2017        NaN   
39377       12345678  control          NaN               2018    manuell   
47598       12345678  control    limousine               2001    manuell   
42221       27322222  control    limousine               2014    manuell   
39705       99999999  control    limousine               1999  automatik   

       power_ps        model  odometer_km  registration_month fuel_type  \
37840       415          911        30000                   7    benzin   
38299         0      x_reihe         5000                   0       NaN   
47337       400      z_reihe        20000                   6    benzin   
12682       500          911         5000                   3    benzin   
35923       400          9

In [18]:
autos = autos[autos["price_dollars"] < 500000]

- There are many luxury cars which explain the higher price. However, many of the cars considered are not luxury post $500,000. Not only this but the prices listed do not appear 'sane'.

In [19]:
autos[['date_crawled','ad_created','last_seen']][0:5]

date_crawled           ad_created            last_seen
0  2016-03-26 17:47:46  2016-03-26 00:00:00  2016-04-06 06:45:54
1  2016-04-04 13:38:56  2016-04-04 00:00:00  2016-04-06 14:45:08
2  2016-03-26 18:57:24  2016-03-26 00:00:00  2016-04-06 20:15:37
3  2016-03-12 16:58:10  2016-03-12 00:00:00  2016-03-15 03:16:28
4  2016-04-01 14:38:50  2016-04-01 00:00:00  2016-04-01 14:38:50

In [20]:
autos['date_crawled'].str[:10].value_counts(normalize=True, dropna=False).sort_index()

2016-03-05    0.025387
2016-03-06    0.013944
2016-03-07    0.035970
2016-03-08    0.033269
2016-03-09    0.033209
2016-03-10    0.032129
2016-03-11    0.032489
2016-03-12    0.036770
2016-03-13    0.015564
2016-03-14    0.036630
2016-03-15    0.033990
2016-03-16    0.029508
2016-03-17    0.031509
2016-03-18    0.013064
2016-03-19    0.034910
2016-03-20    0.037831
2016-03-21    0.037490
2016-03-22    0.032909
2016-03-23    0.032389
2016-03-24    0.029108
2016-03-25    0.031749
2016-03-26    0.032489
2016-03-27    0.031049
2016-03-28    0.034850
2016-03-29    0.034150
2016-03-30    0.033629
2016-03-31    0.031909
2016-04-01    0.033809
2016-04-02    0.035410
2016-04-03    0.038691
2016-04-04    0.036490
2016-04-05    0.013104
2016-04-06    0.003181
2016-04-07    0.001420
Name: date_crawled, dtype: float64

- All information was crawled between March 5th and April 4th 2016.

In [21]:
autos['ad_created'].str[:10].value_counts().sort_index()

2015-06-11       1
2015-08-10       1
2015-09-09       1
2015-11-10       1
2015-12-05       1
2015-12-30       1
2016-01-03       1
2016-01-07       1
2016-01-10       2
2016-01-13       1
2016-01-14       1
2016-01-16       1
2016-01-22       1
2016-01-27       3
2016-01-29       1
2016-02-01       1
2016-02-02       2
2016-02-05       2
2016-02-07       1
2016-02-08       1
2016-02-09       2
2016-02-11       1
2016-02-12       3
2016-02-14       2
2016-02-16       1
2016-02-17       1
2016-02-18       2
2016-02-19       3
2016-02-20       2
2016-02-21       3
              ... 
2016-03-09    1661
2016-03-10    1593
2016-03-11    1639
2016-03-12    1830
2016-03-13     846
2016-03-14    1761
2016-03-15    1687
2016-03-16    1500
2016-03-17    1559
2016-03-18     686
2016-03-19    1692
2016-03-20    1893
2016-03-21    1884
2016-03-22    1638
2016-03-23    1609
2016-03-24    1454
2016-03-25    1594
2016-03-26    1628
2016-03-27    1545
2016-03-28    1748
2016-03-29    1705
2016-03-30  

- The creation of ads is mostly in line with when the information was crawled. This data therefore is heavy on present-day sales.

In [22]:
autos['last_seen'].str[:10].value_counts().sort_index()

2016-03-05       54
2016-03-06      221
2016-03-07      268
2016-03-08      379
2016-03-09      492
2016-03-10      538
2016-03-11      626
2016-03-12     1190
2016-03-13      449
2016-03-14      640
2016-03-15      794
2016-03-16      822
2016-03-17     1396
2016-03-18      371
2016-03-19      787
2016-03-20     1035
2016-03-21     1036
2016-03-22     1079
2016-03-23      929
2016-03-24      978
2016-03-25      960
2016-03-26      848
2016-03-27      801
2016-03-28     1042
2016-03-29     1116
2016-03-30     1242
2016-03-31     1191
2016-04-01     1155
2016-04-02     1244
2016-04-03     1268
2016-04-04     1231
2016-04-05     6212
2016-04-06    11046
2016-04-07     6546
Name: last_seen, dtype: int64

- Again, most ads were last seen during the crawling of the data. Specifically towards the later staging of crawling which implies that many of the cars may still be up for sale even after the data was collected.

In [23]:
autos["registration_year"].describe()

count    49986.000000
mean      2005.075721
std        105.727161
min       1000.000000
25%       1999.000000
50%       2003.000000
75%       2008.000000
max       9999.000000
Name: registration_year, dtype: float64

- The registration of most cards was somewhere in the range of 2003-2005. There is an even number of cars on sale with lower quartile around 1999, the median around 2003, and the upper quartile around 2008. However, a few outliers exist where cars have incorrectly labelled as having registation dates of 1000 or 9999.

In [24]:
autos[autos['registration_year'] < 1950].sort_values(by='registration_year')

date_crawled                                               name  \
22316  2016-03-29 16:56:41              VW_Kaefer.__Zwei_zum_Preis_von_einem.   
49283  2016-03-15 18:38:53                                         Citroen_HY   
24511  2016-03-17 19:45:11                        Trabant__wartburg__Ostalgie   
35238  2016-03-26 13:45:20   Suche_Skoda_Fabia____Skoda_Fabia_Combi_mit_Klima   
10556  2016-04-01 06:02:10                                         UNFAL_Auto   
32585  2016-04-02 16:56:39                                         UNFAL_Auto   
46213  2016-04-02 13:47:16                                      Bellier_Vario   
22659  2016-03-14 08:51:18                                       Opel_Corsa_B   
15898  2016-03-08 10:50:05        Tausch_alles_aus_meinen_Anzeigen_gegen_Auto   
33295  2016-03-12 10:49:42             Ich_verkaufe/suche_fuer_sie_ihr_Auto!!   
42181  2016-03-27 19:50:53          SAMSUNG_55“_3D_Tv_und_Soundbar_gegen_Auto   
3679   2016-04-04 00:36:17                                         Suche_Auto   
45157  2016-03-11 22:37:01                                         Motorhaube   
30781  2016-03-25 13:47:46                   Opel_Calibra_V6_DTM_Bausatz_1:24   
28693  2016-03-22 17:48:41                                     Renault_Twingo   
21416  2016-03-12 08:36:21                            Essex_super_six__Ford_A   
22101  2016-03-09 16:51:17                                         BMW_Andere   
11246  2016-03-26 19:49:59                  Ford_Model_A_Roadster_Deluxe_1931   
2221   2016-03-15 14:57:07  Sehr_seltener_Oldtimer_Opel_1210_zum_Restaurieren   
2573   2016-03-19 22:51:25              Hanomag_rekord_15k_Suche_ersatz_teile   
26607  2016-03-21 14:45:26                                          MG_Andere   
21421  2016-03-05 17:45:32            Ford_Business_Coupe_Hotrod_Projekt.1937   
39725  2016-03-25 22:54:31  Dodge_1937_Hotrod_Flathead_Scheunenfund_Oldtim...   
23804  2016-03-30 22:42:50                                 Mercedes_Benz_170V   
26103  2016-03-16 12:37:38               Oldtimer_Sport___Cabrio_Georges_Irat   
24855  2016-03-31 13:57:18                      Andere_DKW_F8_Cabrio_Limosine   
25792  2016-03-09 10:54:11              Andere_GAZ_Oldtimergelaendewagen_1941   
13963  2016-03-20 17:51:49  Mercedes_Benz_L1500S_Wehrmacht_/_Luftwaffe___F...   
11585  2016-03-11 21:48:36                              Volkswagen__VW_Typ_82   
11047  2016-03-08 20:50:10          Andere_Simca_5_Fourgonette_Kombilimousine   

       price_dollars   abtest vehicle_type  registration_year  gearbox  \
22316           1500  control          NaN               1000  manuell   
49283           7750  control          NaN               1001      NaN   
24511            490  control          NaN               1111      NaN   
35238              0  control          NaN               1500      NaN   
10556            450  control          NaN               1800      NaN   
32585            450  control          NaN               1800      NaN   
46213              0     test   kleinwagen               1910      NaN   
22659            500     test          NaN               1910      NaN   
15898              0     test          NaN               1910      NaN   
33295              0  control          NaN               1910      NaN   
42181              0     test          NaN               1910      NaN   
3679               1     test          NaN               1910      NaN   
45157             15  control          NaN               1910      NaN   
30781             30     test          NaN               1910      NaN   
28693            599  control   kleinwagen               1910  manuell   
21416          16500  control       cabrio               1927  manuell   
22101          11500     test       cabrio               1929  manuell   
11246          27500  control       cabrio               1931  manuell   
2221            3350  control       andere               1934  manuell   
2573            3000  

In [25]:
autos = autos[autos["registration_year"] > 1911]

In [26]:
autos[autos['registration_year'] > 2015].sort_values(by='registration_year')

date_crawled                                               name  \
24746  2016-03-10 17:39:34                                       Opel_Astra_f   
26877  2016-03-14 23:56:11                                  Golf_3_GT_Edition   
26869  2016-03-15 11:51:44         VW_Touran_2_0TDI_DPF_Highline_Standheizung   
26859  2016-03-07 13:36:21                 Opel_Astra__1.6_benzin_TÜV_07.2016   
26809  2016-03-14 18:30:40                          BMW_E39_520i___nur_60_TKM   
26767  2016-03-10 09:46:29  Opel_Corsa_c_1.2_16v_Eco_Auto_wird_noch_gefahr...   
26764  2016-03-26 22:49:57             Nissan.micra_1.0._servo_zentral_klima_   
26745  2016-03-31 09:36:19    Renault_Clio_1_2l_Klima__4_Tueren_Laufl.__83Tkm   
26655  2016-03-21 16:59:02           VW_Passat_1.8_Automatik_Tuev_Beschaedigt   
26631  2016-04-04 02:02:15                              Fahrzeug_zu_verkaufen   
26619  2016-03-24 20:47:50                       Kleiner_Diesel_mit_TÜV_11/16   
26614  2016-03-22 10:57:42                                         Vw_polo_6n   
26547  2016-03-12 19:46:27                       Alfa_Rome_147_Eco_Bj_09/2003   
26478  2016-03-14 22:55:03                                         Audi_80_B4   
26474  2016-03-11 14:46:20           BMW_E39_520i_173tkm_auch_Tausch_moeglich   
26375  2016-04-01 17:56:59                             Ford_Focus_Diesel__1.8   
26361  2016-03-10 12:50:19  Suzuki_Swift_1_0_Automatik_TÜV_10/2016_erst_10...   
26357  2016-03-09 12:59:01                        Daimler_Chrysler_PT_Cruiser   
26356  2016-03-16 19:49:18                       Nissan_Micra_K11_12.16_TÜV!!   
26339  2016-03-12 23:51:23                     Vw_sharan_2.0_benzin_lpg_2hand   
26267  2016-03-26 11:50:09                                        Ford_Galaxy   
26236  2016-03-25 10:53:26                  Verkaufe_citroen_saxo_1.0_mit_TÜV   
26162  2016-03-23 17:48:52                                     Opel_corsa_1.0   
25915  2016-03-19 14:36:25                                   Mercedes_200_cdi   
25909  2016-03-07 17:41:54                      Corsa_a_Faehrt__Bremst__Lenkt   
25883  2016-03-08 16:57:19                                      Ford_ka_Sport   
26900  2016-03-15 13:59:39                              Mercedes_Sprinter_208   
25878  2016-04-01 15:47:41          Ford_Fiesta_1_6l_super_Auto_technisch_top   
26941  2016-03-21 13:41:49                  BMW_120_DIESEL_/_184_PS_/_SPARSAM   
27137  2016-03-22 22:51:57      BMW_740Li_Facelift_Modell_V8_NOTVERKAUF_8900€   
...                    ...                                                ...   
4359   2016-03-17 21:53:22                                       Bmw_e39_523i   
25318  2016-03-05 20:52:07                                       Auto_gesucht   
23314  2016-03-26 22:38:19  Gepflegter_VW_Touran_1.9TDI_Standheizung_Klima...   
23227  2016-03-28 15:45:05                           Seat_Leon_Refference_1.6   
11290  2016-03-26 19:52:36                                  Mazda_323F_Sporty   
40254  2016-03-17 17:54:53                                Mercedes_Benz_E_240   
4241   2016-03-25 18:53:16  Hyndai_Lantra_1.6l_/_TÜV_NEU__03.2018_/_Klima_...   
40078  2016-03-29 01:36:22  Lancia_ypsilon_1.4__klima/_2_Hand_Neue_/TÜVSch...   
24834  2016-03-22 22:49:54             Audi_80_B4_2.0_Top_Zustand!!_TÜV_neu!!   
38342  2016-03-07 18:52:22              Suche_Auto_fuer_die_jagd_zum_tauschen   
49185  2016-03-28 11:45:31              Alfa_Romeo_156_Sportwagon_1_8_T_Spark   
5763   2016-03-05 19:54:49             Suche_Auspuff_Anlage_honda_crx_del_sol   
27578  2016-03-31 22:49:04                                   VW_GOLF_2_SYNCRO   
4549   2016-04-01 21:57:05                                         Kompressor   
453    2016-03-28 13:51:12                                         Armee_Jeep   
42079  2016-03-17 19:55:29                      APE_50___deutsche_Papiere_!!!   
4164   2016-03-29 18:39:40  Verkaufe_DESIGN_Streifen_/_Aufkleber_VW__Opel_...   
22799  2016-03-20 18:56:44                   

In [27]:
autos = autos[autos["registration_year"] < 2018]

- After reviewing the data. While it is somewhat plausabile that some cars were registered on or before 1910, there are many inconsistencies with the data. 
- The same can be said with cars registered from 2018 onwards. Some cars registered after 2016, may exist but have been incorrectly labelled. However, again, there are many inconsistencies to allow this information.

In [28]:
autos['brand'].value_counts()

volkswagen        10563
opel               5388
bmw                5382
mercedes_benz      4689
audi               4254
ford               3449
renault            2367
peugeot            1449
fiat               1287
seat                915
skoda               782
mazda               751
nissan              749
smart               695
citroen             692
toyota              615
sonstige_autos      526
hyundai             485
volvo               454
mini                424
mitsubishi          399
honda               391
kia                 353
alfa_romeo          327
porsche             293
suzuki              289
chevrolet           279
chrysler            181
dacia               128
daihatsu            126
jeep                109
subaru              107
land_rover           99
daewoo               78
saab                 78
jaguar               76
trabant              76
rover                69
lancia               56
lada                 29
Name: brand, dtype: int64

- The top five manufacturers on the list are all German brands with Volkswagen by far the most popular.
- We'll limit our aggregate analysis to brands in the top 20 car sales represented in our data set.

In [35]:
brand_list = autos['brand'].value_counts()
brand_top_10 = brand_list[:10].index
brand_top_10

Index(['volkswagen', 'opel', 'bmw', 'mercedes_benz', 'audi', 'ford', 'renault',
       'peugeot', 'fiat', 'seat'],
      dtype='object')

In [36]:
brand_mean_prices = {}
for brand in brand_top_10:
    brand_mean_prices[brand] = autos[autos['brand'] == brand]['price_dollars'].mean()
    
brand_mean_prices

{'audi': 8989.552891396334,
 'bmw': 8051.632664437012,
 'fiat': 2693.4957264957266,
 'ford': 3632.7184691214843,
 'mercedes_benz': 8384.584346342504,
 'opel': 2854.7763548626576,
 'peugeot': 3014.3381642512077,
 'renault': 2365.113645965357,
 'seat': 4259.546448087432,
 'volkswagen': 5177.634857521537}

- Based on the results of our aggregrate data, it is clear that there are around  3 tiers interms of car sale prices.
- The first tier appears to be anything over \$8000 including Audi, BMW and Mercedes_Benz. This is to be expected as these are considered luxury car brands.
- The more common and affordable car brands such as Fiat, Ford, Opel, Peugeot, and Renault occupy the lower tier where cars cost anything lower than \$4000
- There also appears to be a middle tier around the \$5000 mark which is occupied by Volkswagen. Volkswagen has the largest number of data in our dataset, and it it can be assumed that this is a realistic result.
- Seat cars appear to be selling just around \$4250 which I would say is also close to this middle-tier and not a lower-tier car price as there is huge seperation between it and other lower-tier brands.

In [37]:
brand_top_5 = brand_list[:5].index

brand_mean_prices = {}
for brand in brand_top_5:
    brand_mean_prices[brand] = autos[autos['brand'] == brand]['price_dollars'].mean()
    
brand_mean_mileage = {}
for brand in brand_top_5:
    brand_mean_mileage[brand] = autos[autos['brand'] == brand]['odometer_km'].mean()

In [39]:
brand_mean_prices

{'audi': 8989.552891396334,
 'bmw': 8051.632664437012,
 'mercedes_benz': 8384.584346342504,
 'opel': 2854.7763548626576,
 'volkswagen': 5177.634857521537}

In [40]:
brand_mean_mileage

{'audi': 129561.58909261871,
 'bmw': 132517.6514306949,
 'mercedes_benz': 131037.53465557688,
 'opel': 129324.42464736452,
 'volkswagen': 128970.46293666572}

In [42]:
mean_price = pd.Series(brand_mean_prices)
mean_mileage = pd.Series(brand_mean_mileage)

mean_price

audi             8989.552891
bmw              8051.632664
mercedes_benz    8384.584346
opel             2854.776355
volkswagen       5177.634858
dtype: float64

In [43]:
mean_mileage

audi             129561.589093
bmw              132517.651431
mercedes_benz    131037.534656
opel             129324.424647
volkswagen       128970.462937
dtype: float64

In [45]:
top_5_brands_info = pd.DataFrame(mean_price, columns=['mean_price'])
top_5_brands_info

mean_price
audi           8989.552891
bmw            8051.632664
mercedes_benz  8384.584346
opel           2854.776355
volkswagen     5177.634858

In [47]:
top_5_brands_info['mean_mileage'] = mean_mileage
top_5_brands_info

mean_price   mean_mileage
audi           8989.552891  129561.589093
bmw            8051.632664  132517.651431
mercedes_benz  8384.584346  131037.534656
opel           2854.776355  129324.424647
volkswagen     5177.634858  128970.462937

- Generally, the mean mileage for the top brands in our dataset sit around 130,000km. This affects all of our top brands of course. However, the mean price for each car differs depending on car bran tier